File: client/src/types/integrations.ts
export interface IntegrationStatus {
  provider: string
  is_connected: boolean
  is_enabled: boolean
  external_user_id: string | null
  display_name: string | null
  connected_at: string | null
}

export interface IntegrationListResponse {
  integrations: IntegrationStatus[]
}

export interface TelegramConnectResponse {
  deep_link: string
  token: string
  expires_in_seconds: number
}

export interface TelegramDisconnectResponse {
  success: boolean
  message: string
}

File: server/src/schemas/integrations.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class IntegrationStatus(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    provider: str
    is_connected: bool
    is_enabled: bool = False
    external_user_id: Optional[str] = None
    display_name: Optional[str] = None
    connected_at: Optional[datetime] = None


class IntegrationListResponse(BaseModel):
    integrations: list[IntegrationStatus]


class TelegramConnectResponse(BaseModel):
    deep_link: str
    token: str
    expires_in_seconds: int


class TelegramDisconnectResponse(BaseModel):
    success: bool
    message: str
